pkg/secrets/provide_conjur_secrets.go

```diff
@@ -97,3 +96,55 @@ func RetryableSecretProvider(
 		return err
 	}
 }
+
+var TickerQuit = make(chan bool)
+var ProviderDone = make(chan bool)
+var SecretProviderError = make(chan error)
+func PeriodicSecretProvider(
+	secretRefreshInterval time.Duration,
+	mode string,
+	provideSecrets ProviderFunc,
+	) ProviderFunc {
+	return func() error {
+		err := provideSecrets()
+		if err != nil {
+			SecretProviderError <- err
+			return err
+		}
+		switch {
+		case mode != "sidecar":
+			SecretProviderError <- err
+			return err
+		case mode == "sidecar" && secretRefreshInterval > 0:
+			ticker := time.NewTicker(secretRefreshInterval)
+			go periodicSecretProvider(provideSecrets, ticker)
+		default:
+			// fall through
+		}
+		// Wait here
+		select {
+		case <-ProviderDone: // Todo change to SIGTERM or SIGINTERRUPT
+		}
+		if mode == "sidecar" && secretRefreshInterval > 0{
+			TickerQuit <- true
+			time.Sleep(1 * time.Second) // give a second for the go routine to shut down
```

Comment:
Might be better to define this as a const at the top of the file so it's easy to find/change.